tree/ntuple/v7/inc/ROOT/RNTupleJoinTable.hxx

```diff
@@ -138,14 +191,46 @@ public:
    /// Only built join tables can be queried.
    bool IsBuilt() const { return fIsBuilt; }
 
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Get the partition keys used in the join table.
+   const std::vector<PartitionKey_t> &GetPartitionKeys() const { return fPartitionKeys; }
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Get all entry indexes for the given join field value(s) within a set of partitions.
+   ///
+   /// \param[in] valuePtrs A vector of pointers to the join field values to look up.
+   /// \param[in] partitionKeys The partition keys to use for the lookup.
+   ///
+   /// \return The entry numbers that correspond to `valuePtrs`. When there are no corresponding entries, an empty
+   /// vector is returned.
+   std::vector<ROOT::NTupleSize_t>
+   GetEntryIndexes(const std::vector<void *> &valuePtrs, const std::vector<PartitionKey_t> &partitionKeys) const;
+
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Get all entry indexes for the given join field value(s) within a partition.
+   ///
+   /// \param[in] valuePtrs A vector of pointers to the join field values to look up.
+   /// \param[in] partitionKey The partition key to use for the lookup.
+   ///
+   /// \return The entry numbers that correspond to `valuePtrs`. When there are no corresponding entries, an empty
+   /// vector is returned.
+   std::vector<ROOT::NTupleSize_t>
+   GetEntryIndexes(const std::vector<void *> &valuePtrs, const PartitionKey_t &partitionKey) const
+   {
+      return GetEntryIndexes(valuePtrs, std::vector{partitionKey});
+   }
+
    /////////////////////////////////////////////////////////////////////////////
    /// \brief Get all entry indexes for the given join field value(s).
    ///
    /// \param[in] valuePtrs A vector of pointers to the join field values to look up.
    ///
-   /// \return The entry indexes that correspond to `valuePtrs`. An empty vector is returned when there are no matching
-   /// indexes.
-   std::vector<ROOT::NTupleSize_t> GetEntryIndexes(const std::vector<void *> &valuePtrs) const;
+   /// \return The entry numbers that correspond to `valuePtrs`. When there are no corresponding entries, an empty
+   /// vector is returned.
+   std::vector<ROOT::NTupleSize_t> GetEntryIndexes(const std::vector<void *> &valuePtrs) const
+   {
+      return GetEntryIndexes(valuePtrs, fPartitionKeys);
+   }
```

Comment:
Is this useful in case of multiple partitions since the return value loses the information which mapping the indexes are relative to? Maybe what we want instead is `GetEntryIndexes(const std::vector<void *> &valuePtrs, PartitionKey_t partitionKey = kDefaultPartitionKey)`?